packages/material-ui/src/Chip/Chip.js

```diff
@@ -1,23 +1,101 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
+import { deepmerge } from '@material-ui/utils';
+// import { useThemeVariants } from '@material-ui/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import CancelIcon from '../internal/svg-icons/Cancel';
-import withStyles from '../styles/withStyles';
 import { emphasize, alpha } from '../styles/colorManipulator';
 import useForkRef from '../utils/useForkRef';
 import unsupportedProp from '../utils/unsupportedProp';
 import capitalize from '../utils/capitalize';
 import ButtonBase from '../ButtonBase';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import chipClasses, { getChipUtilityClass } from './chipClasses';
 
-export const styles = (theme) => {
-  const backgroundColor =
-    theme.palette.mode === 'light' ? theme.palette.grey[300] : theme.palette.grey[700];
-  const deleteIconColor = alpha(theme.palette.text.primary, 0.26);
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+  const { color, clickable, onDelete, size } = styleProps;
 
-  return {
-    /* Styles applied to the root element. */
-    root: {
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.variant],
+    ...(styleProps.disabled && styles.disabled),
```

Comment:
```suggestion
```

No overrides for pseudo selectors. Developers must the the pseudo classes, `.Mui-disabled` in this case